### DISCUSSION TITLE
Enhancement: azure deployment with active directory as auth provider

### DISCUSSION THREAD

**Discussion by thunderbug1** [danny-avila/LibreChat#515] [Feature Requests & Suggestions]
### What features would you like to see added?

I would like to deploy it to azure and use active directory as the auth provider. As far as I can see that is not supported out of the box. I can help to implement this and would like to start a discussion about how to best integrate this functionality.

### More details

~~this probably will also require a feature toggle for local sign up since active directory would handle the user management.~~

### Which components are impacted by your request?

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
@thunderbug1 Hi, thanks for your willingness on this! A user is working on this in #481, you can take a look at that PR which will be revisited after they clean up some things.

On deploying, I've heard of people doing it successfully through Azure but no one has contributed notes on this, that would be helpful, too if you were willing

**Comment by danny-avila** [comment]
Active work on this migrated to #495

**Comment by thunderbug1** [comment]
yes thank you, I saw it.
It seems pretty completed to me. Do you need help testing it?

**Comment by danny-avila** [comment]
> yes thank you, I saw it. It seems pretty completed to me. Do you need help testing it?

feel free to test it! I'm not actively developing it, and don't have an azure AD

**Comment by edgBR** [comment]
Hi @danny-avila 

The integration for oauth with entra ID is working properly for me but I am wandering what about the Azure Open AI Endpoints.

Currently it seems that is needed to provide an API key.
